### PR TITLE
[SonyLIV] Add SonyLIVSeriesIE.

### PR DESCRIPTION
**Pull request by Ashish0804**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

Added support to download whole series/shows from sonyliv.

### CODE REVIEW

File: yt_dlp/extractor/sonyliv.py [hunk 1]
```diff
@@ -107,3 +114,35 @@ def _real_extract(self, url):
...
+        headers = {
+            'Accept': 'application/json, text/plain, */*',
+            'Referer': 'https://www.sonyliv.com',
+        }
+        headers['security_token'] = self._download_json(self._API_SECURITY_URL, video_id=show_id, headers=headers, note='Downloading security token')['resultObj']
+        show_json = self._download_json(self._API_SHOW_URL.format(show_id), video_id=show_id, headers=headers)
+        for season in try_get(show_json, lambda x: x['resultObj']['containers'][0]['containers'], list) or []:
+            season_id = season['id']
+            episodes_json = self._download_json(self._API_EPISODES_URL.format(season_id), video_id=season_id, headers=headers,)
+            for episode in try_get(episodes_json, lambda x: x['resultObj']['containers'][0]['containers'], list) or []:
```
Comment (pukkandan) [inline yt_dlp/extractor/sonyliv.py:142]:
```suggestion
         headers['security_token'] = self._download_json(
            self._API_SECURITY_URL, video_id=show_id, headers=headers,
            note='Downloading security token')['resultObj']
        seasons = try_get(
            self._download_json(self._API_SHOW_URL.format(show_id), video_id=show_id, headers=headers),
            lambda x: x['resultObj']['containers'][0]['containers'], list)
        for season in seasons or []:
            season_id = season['id']
            episodes = try_get(
                self._download_json(self._API_EPISODES_URL.format(season_id), video_id=season_id, headers=headers),
                lambda x: x['resultObj']['containers'][0]['containers'], list)
            for episode in episodes or []:
```

**Review by pukkandan — APPROVED** [review]
a lot of the lines are too long. while not a strict requirement, try to keep each line 100-120 chars atmost

File: yt_dlp/extractor/sonyliv.py [hunk 2]
```diff
@@ -107,3 +114,41 @@ def _real_extract(self, url):
...
+    _API_SHOW_URL = "https://apiv2.sonyliv.com/AGL/1.9/R/ENG/WEB/IN/DL/DETAIL/{}?kids_safe=false&from=0&to=49"
+    _API_EPISODES_URL = "https://apiv2.sonyliv.com/AGL/1.4/R/ENG/WEB/IN/CONTENT/DETAIL/BUNDLE/{}?from=0&to=1000&orderBy=episodeNumber&sortOrder=asc"
+    _API_SECURITY_URL = 'https://apiv2.sonyliv.com/AGL/1.4/A/ENG/WEB/ALL/GETTOKEN'
+
+    def _entries(self, show_id):
+        headers = {
+            'Accept': 'application/json, text/plain, */*',
+            'Referer': 'https://www.sonyliv.com',
+        }
+         headers['security_token'] = self._download_json(
```
Comment (pukkandan) [inline yt_dlp/extractor/sonyliv.py:137]:
```suggestion
        headers['security_token'] = self._download_json(
```

**Comment by Ashish0804** [comment]
> a lot of the lines are too long. while not a strict requirement, try to keep each line 100-120 chars atmost

Got it.
I'll keep it in mind next time.

**Pull request merged**